app/assets/javascripts/discourse/app/lib/url.js

```diff
@@ -238,7 +238,7 @@ const DiscourseURL = EmberObject.extend({
     const m = /^#(.+)$/.exec(path);
     if (m) {
       jumpToElement(m[1]);
-      return this.replaceState(path);
+      return this.replaceState(`${this.router.currentURL}${path}`);
```

Comment:
Is it possible for `this.router.currentURL` to already have a fragment at the end? If so, might we end up with `/t/123#oldfragment#newfragment` ?